backend/services/src/main/java/com/crimealert/services/UserService.java

```diff
@@ -125,6 +126,75 @@ public String deleteProfile(String email)
 		
 	}
 	
+	public Document insertOtp(Otp otp)
+	{
+		Document document = new Document();
+		try {
+
+        	MongoClient mongoClient = getDBConnectionService().getDBConnection();
+            
+            MongoDatabase database = mongoClient.getDatabase(UserConstant.DB);
+            
+        	//Preparing a document
+            
+            document.append(UserConstant.EMAIL, otp.getEmail());
+            document.append(UserConstant.TOKEN, otp.getOneTimeToken());
+            
+            //Inserting the document into the collection
+            database.getCollection(UserConstant.OTP_COLLECTION).insertOne(document);
+            
+            getDBConnectionService().closeDBConnection();
+            
+            System.out.println("Document inserted successfully");
+        } catch (MongoException me) {
+            System.err.println("An error occurred while attempting to run a command: " + me);
+            throw me;
+        }
+        catch (ClientSideException ce) {
+            System.err.println("An error occurred while attempting to run a command: " + ce);
+            throw ce;
+        }
+		
+		return document;
+		
+	}
+	
+	public long deleteOtp(Otp otp)
+	{
+		try {
+
+        	MongoClient mongoClient = getDBConnectionService().getDBConnection();
+            
+            MongoDatabase database = mongoClient.getDatabase(UserConstant.DB);
+            
+            //Inserting the document into the collection
+            Bson filter = Filters.and(Filters.eq(UserConstant.EMAIL, otp.getEmail()), Filters.eq(UserConstant.TOKEN, otp.getOneTimeToken()));
```

Comment:
Can we reuse searchOtp here?

Reply:
Delete needs filter and not document :p buhahaha